### PR TITLE
Allow templating access to request body for use with jsonpath and xpath

### PR DESCRIPTION
**Pull request by benjih**
Also some functional tests and also request method as it was very easy. https://github.com/SpectoLabs/hoverfly/issues/678

This PR also changes XML parsing to be not strict, which should fix https://github.com/SpectoLabs/hoverfly/issues/744

### CODE REVIEW

File: core/templating/templating.go [hunk 1]
```diff
@@ -47,6 +50,7 @@ func NewTemplator() *Templator {
 		raymond.RegisterHelper("randomIPv4", t.randomIPv4)
 		raymond.RegisterHelper("randomIPv6", t.randomIPv6)
 		raymond.RegisterHelper("randomUuid", t.randomUuid)
+		raymond.RegisterHelper("Request.body jsonPath", t.jsonPath)
```
Comment (tommysitu) [inline core/templating/templating.go]: Is it supposed to register {{ Request.Body xpath }} as well?

File: functional-tests/testdata/templating_helpers.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
+package testdata
+
+var TemplatingHelpers = `{
```
Comment (tommysitu) [inline functional-tests/testdata/templating_helpers.go]: I don't think the test data is being used.

Comment (benjih) [reply functional-tests/testdata/templating_helpers.go]: https://github.com/SpectoLabs/hoverfly/blob/templating-request-body/functional-tests/core/ft_templated_response_test.go#L120

File: core/templating/templating.go (hunk 1 above)
Comment (benjih) [reply core/templating/templating.go]: This was an old attempt, its not used or required, should be deleted.

**Pull request merged**